src/main/java/org/folio/service/job/JobExecutionServiceImpl.java

```diff
@@ -186,7 +192,15 @@ private void prepareJobExecution(JobExecution jobExecution, FileDefinition fileE
           List<JobExecution> jobExecutionList = asyncResult.result();
           jobExecutionList.forEach(jobExe -> {
             jobExe.setStatus(FAIL);
+            //reset progress to skip already exported/failed records
+            if (jobExe.getProgress() != null) {
+              Progress jobProgress = jobExe.getProgress();
+              jobProgress.setFailed(0);
+              jobProgress.setExported(0);
+              jobProgress.setTotal(0);
```

Comment:
Please use a chain of .withXXX() methods instead of setters

Reply:
Fixed, thank you.